### DISCUSSION TITLE
How to enable hardware virtualization in github self-hosted action runners?

### DISCUSSION THREAD

**Discussion by koteswara-kelam** [community/community#26216] [Actions]
<p>I want to run my CI job in github self-hosted action runner(linux) but that job needs hardware virtualization to be enabled. Is there any way to enable hardware virtualization in self-hosted runner? In other words, can I create a vm inside self-hosted runner?</p>

**Comment by Yanjingzhu** [comment]
<p>Hi, </p>
<p>What’s your meaning of self-hosted action runners? Is it the same conception with self-hosted runner which introduced in the document: <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-self-hosted-runners" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-self-hosted-runners</a></p>
<p>What’s your workflow yml definition? Do you get any error message when run it? </p>

**Comment by koteswara-kelam** [comment]
<p>Thanks for reply. Sorry, I misunderstood self-hosted runners as github-hosted runners. I want to add self-hosted runner to github repository. So I downloaded actions-runner package on my baremetal mechine(ubuntu 16.04) and tried to run config.sh but it fails with following error:</p>
<p>~/actions-runner$ ./config.sh --url <a href="https://github.com/koti-test-repo/testing-actions" rel="nofollow noopener">https://github.com/koti-test-repo/testing-actions</a> --token ABVAMURGO57U7YY5TIUPSXC526ODC</p>
<p>…</p>
<p>…<br>
| Self-hosted runner registration |</p>
<div class="md-table">
<table>
<thead>
<tr>
<th></th>
</tr>
</thead>
<tbody></tbody>
</table>
</div><h1>Authentication</h1>
<p>The operation was canceled.</p>
<p>Did I miss anything here? how to debug this issue?</p>

**Comment by Yanjingzhu** [comment]
<p>You can find the runner logs folder _diag under actions-runner folder. You can provide logs here for us to investigate. </p>
<p><img src="https://user-images.githubusercontent.com/24806478/181096103-3bfc95f5-4e9e-468b-bb6e-71254c8aefd9.png" alt="self-runner _diag.png" data-base62-sha1="kt61GqTqNUsXQTYQUthKGZWwArb" width="687" height="135"></p>
<p>The token show in configure script is an automatically-generated time-limited token to authenticate the request. Did you configure the runner in time after you downloaded actions-runner package?</p>
<p>There is the document for Adding self-hosted runners: </p>
<p><a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/adding-self-hosted-runners" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/adding-self-hosted-runners</a></p>

**Comment by koteswara-kelam** [comment, marked as answer]
<p>I was able to solve the issue by adding proxy url in “.proxy” file. Thanks.</p>

**Comment by donovanmuller** [comment]
<p>More info here: <a href="https://github.com/microsoft/azure-pipelines-agent/blob/master/docs/start/proxyconfig.md" rel="nofollow noopener">https://github.com/microsoft/azure-pipelines-agent/blob/master/docs/start/proxyconfig.md</a></p>
<p>Seems to be a striking resmblence here :wink:</p>

**Comment by dralley** [comment]
<p>Speaking of which, this is enabled on Travis natively, but not on Github Actions.  We’d really like to test our selinux configuration but we can’t do so easily on Github Actions due to KVM hardware nested virt not being enabled, at least not without doing some self-hosting.  Any chance this will change?</p>
